src/linters/stylelint/index.js

```diff
@@ -0,0 +1,111 @@
+const Module = require( 'module' );
+const path = require( 'path' );
+const moduleAlias = require( 'module-alias' );
+
+/**
+ * Convert an error message from the stylelint format into one acceptable for GitHub
+ *
+ * @param {Object} message Data about a warning from stylelint.
+ * @returns {Object}
+ */
+const formatMessage = message => ( {
+	line: message.line,
+	column: message.column,
+	severity: message.severity,
+	message: message.text,
+	source: message.rule,
+} );
+
+/**
+ * Fetch a count of the total errors and warnings for our response.
+ *
+ * @param {Object} files Formatted warnings against specific files.
+ * @returns {{warnings: number, errors: number}}
+ */
+const getTotals = ( files ) => {
+	const allErrors = Object.keys( files ).reduce( ( acc, key ) => [ ...acc, ...files[key] ], [] );
+
+	return {
+		errors: allErrors.filter( errorData => errorData.severity === 'error' ).length,
+		warnings: allErrors.filter( errorData => errorData.severity === 'warning' ).length,
+	};
+};
+
+/**
+ * Properly Format the return for GitHub.
+ *
+ * @param {Object} data Raw data from stylelint Node runner.
+ * @param {String} codepath Path against which to check files.
+ * @returns {{files: {}, totals: {warnings: number, errors: number}}}
+ */
+const formatOutput = ( data, codepath ) => {
+	const files = {};
+	data.results.forEach( result => {
+		// Only parse through CSS or SCSS files.
+		if ( ! result.source.match( /\.s?css$/ ) ) {
+			return;
+		}
+
+		// Exclude any empty files.
+		if ( ! result.warnings.length ) {
+			return;
+		}
+
+		const relPath = path.relative( codepath, result.source );
+		files[ relPath ] = result.warnings.map( formatMessage );
+	} );
+
+	return { totals: getTotals( files ), files };
```

Comment:
I guess this should maybe be multiple lines too
```suggestion
	return {
		totals: getTotals( files ),
		files,
	};
```